### PR TITLE
Issue 2223 pro cyclic multi sm validator

### PR DESCRIPTION
**Pull request by rherrmannr**

### CODE REVIEW

File: plugins/org.yakindu.sct.model.sgraph/src/org/yakindu/sct/model/sgraph/util/DFS.java [hunk 1]
```diff
@@ -90,10 +92,13 @@ public synchronized void perform(Object rootElement) {
 
 
 	/** Visits an element. */
-	private int visit(Object element, Object parent, int depth) {
+	public int visit(Object element, Object parent, int depth) {
```
Comment (tkutz) [inline plugins/org.yakindu.sct.model.sgraph/src/org/yakindu/sct/model/sgraph/util/DFS.java]: Where is this required?

File: plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend [hunk 2]
```diff
@@ -791,7 +791,15 @@ class STextValidator extends AbstractSTextValidator implements STextValidationMe
 			}
 		}
 	}
-	
+	@Check
+	def void checkImportNotCyclicsct(Statechart sct) {
```
Comment (tkutz) [inline plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend]: Why is there an "sct" in the name? Looks strange.

File: plugins/org.yakindu.sct.model.sgraph/src/org/yakindu/sct/model/sgraph/util/DFS.java (hunk 1 above)
Comment (rherrmannr) [reply plugins/org.yakindu.sct.model.sgraph/src/org/yakindu/sct/model/sgraph/util/DFS.java]: it isn't.

File: plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend (hunk 2 above)
Comment (rherrmannr) [reply plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend]: renamed

Comment (tkutz) [reply plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend]: I meant checkImportNotCyclic**sct** => checkImportNotCyclic

Comment (rherrmannr) [reply plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend]: :+1:

**Pull request merged**